app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find_by(id: params[:user_id])
-    @posts = @user.posts.order(id: :asc)
+    @user = @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:user).where(user_id: params[:user_id]).references(:posts)
   end
 
   def show
     @post = Post.find(params[:id])
+    @comments = @post.comments.includes(:user)
   end
 
   def create
```

Comment:
- In order to test the n+1 solution, I would recommend fixing (syntax error) this issue which is happening while we are creating the posts.

![image](https://user-images.githubusercontent.com/55635977/192633987-61eaae2a-f8d6-4975-88bf-3bc1d16f4aa5.png)